x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/util/general.ts

```diff
@@ -63,18 +67,9 @@ function getDetectors(job: Job, datafeed: Datafeed) {
     }
   } else {
     // all other detectors.
+    detectors = processFieldlessAggs(detectors);
```

Comment:
For indexes with no numeric fields, we still allow the user to pick functions which require a field e.g. `mean` :

![image](https://user-images.githubusercontent.com/7405507/66747089-92099380-ee7b-11e9-995c-73eeaf1e2729.png)

The old wizard let you do e.g. `mean` on non-numeric fields, so I think this behaviour is better in some ways - as only numeric fields appear in the 'Field' dropdown, but it would be better still if we prevented the user from selecting such metrics.

Reply:
Yes, that would be a nice improvement. i've added an item for it in the project https://github.com/elastic/kibana/projects/30#card-27745649

Reply:
Agree that this should go in a follow-up. Currently you can go all the way through the wizard and create the job. Only when the job is started do you see an error:

![image](https://user-images.githubusercontent.com/7405507/66766937-35bc6900-eea7-11e9-817c-4ec03de38ac6.png)

For indexes with numeric fields, the Save button in the Add Detector is disabled until you have selected the metric field.